libraries/AP_ESC_Telem/AP_ESC_Telem_Backend.h

```diff
@@ -0,0 +1,64 @@
+#pragma once
+
+#include <AP_HAL/AP_HAL.h>
+
+#ifndef HAL_WITH_ESC_TELEM
+#define HAL_WITH_ESC_TELEM 1
+#endif
+
+#if HAL_WITH_ESC_TELEM
+
+class AP_ESC_Telem;
+
+class AP_ESC_Telem_Backend {
+public:
+
+    struct TelemetryData {
+        int16_t  temperature_deg;   // degrees C, negative values allowed
+        uint16_t voltage_cv;        // centi-volts
+        uint16_t current_ca;        // centi-amps
+        uint16_t consumption_mah;   // mAh
+        uint32_t usage_s;           // usage seconds
+        int16_t  motor_temp_deg;    // degrees C, negative values allowed
+        uint32_t last_update_ms;    // last update time, determines whether active
+        uint8_t  types;             // telemetry types present
```

Comment:
I think having this uint8 here, mixes with the uint16 and uint32 will cause some unaligned accesses, preventing "atomic" updates of these structure variables